projects/packages/publicize/src/rest-api/class-share-post-controller.php

```diff
@@ -0,0 +1,142 @@
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->base_api_path = 'wpcom';
+		$this->version       = 'v2';
+
+		$this->namespace = "{$this->base_api_path}/{$this->version}";
+		$this->rest_base = 'publicize/share-post';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register the routes.
+	 */
+	public function register_routes() {
+
+		$args = array(
+			'methods'             => WP_REST_Server::CREATABLE,
+			'callback'            => array( $this, 'share_post' ),
+			'permission_callback' => array( $this, 'permissions_check' ),
+			'args'                => array(
+				'message'             => array(
+					'description'       => __( 'The message to share.', 'jetpack-publicize-pkg' ),
+					'type'              => 'string',
+					'required'          => true,
+					'validate_callback' => function ( $param ) {
+						return is_string( $param );
+					},
+					'sanitize_callback' => 'sanitize_textarea_field',
+				),
+				'skipped_connections' => array(
+					'description'       => __( 'Array of external connection IDs to skip sharing.', 'jetpack-publicize-pkg' ),
+					'type'              => 'array',
+					'required'          => false,
+					'validate_callback' => function ( $param ) {
+						return is_array( $param );
+					},
+					'sanitize_callback' => function ( $param ) {
+						return array_map( 'absint', $param );
+					},
+				),
+				'async'               => array(
+					'description' => __( 'Whether to share the post asynchronously.', 'jetpack-publicize-pkg' ),
+					'type'        => 'boolean',
+					'default'     => false,
+				),
+			),
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<postId>\d+)',
+			$args
+		);
+
+		if ( Utils::is_wpcom() ) {
+			// WPCOM Legacy route for backwards compatibility.
+			// TODO: Remove this after April 2025 release of Jetpack.
+			register_rest_route(
+				$this->namespace,
+				'/posts/(?P<postId>\d+)/publicize',
+				$args
+			);
+		}
+	}
+
+	/**
+	 * Ensure the user has proper tokens and permissions to publish posts on this blog.
+	 *
+	 * @return WP_Error|boolean
+	 */
+	public function permissions_check() {
+		return $this->publicize_permissions_check();
+	}
+
+	/**
+	 * If this method callback is executed on WPCOM, we share the post using republicize_post(). If this method callback
+	 * is executed on a Jetpack site, we make an API call to WPCOM using wpcom_json_api_request_as_user() and return
+	 * the results. In both cases, this file and method are executed, as this file is synced from Jetpack to WPCOM.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error The publicize results, including two arrays: `results` and `errors`
+	 */
+	public function share_post( $request ) {
+		$post_id = $request->get_param( 'postId' );
+
+		if ( Utils::is_wpcom() ) {
```

Comment:
This is a nitpick, but I'd prefer this check to be negated, so we can proxy and return earlier as it would reduce the nesting.

I've also noticed this pattern fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qcyhtva%2Sfha%2S_vap%2Syvo%2Spber%2Qncv%2Sjcpbz%2Qraqcbvagf%2Spynff%2Qjcpbz%2Qerfg%2Qncv%2Qi2%2Qraqcbvag%2Qarjfyrggre%2Qpngrtbevrf%2Qfhofpevcgvbaf%2Qpbhag.cuc%3Se%3Q8sn0rqr6%2340%2Q43-og which means we don't get here unless we're on WPCOM

Reply:
> This is a nitpick, but I'd prefer this check to be negated, so we can proxy and return earlier as it would reduce the nesting.

Yeah, sounds good. Thanks

> I've also noticed this pattern ... which means we don't get here unless we're on WPCOM

We can't use that when you have to pass that `$path` parameter to `proxy_request_to_wpcom_as_user`